mne/preprocessing/eyetracking/eyetracking.py

```diff
@@ -164,3 +170,137 @@ def _convert_mm_to_m(array):
 
 def _convert_deg_to_rad(array):
     return array * np.pi / 180.0
+
+
+def convert_units(inst, calibration, to="radians"):
+    """Convert Eyegaze data from pixels to radians of visual angle or vice versa.
+
```

Comment:
We should have a prominent warning at the top like:
```suggestion

    .. warning::
        Currently, eyegaze channel units of radians not be reported correctly in most
        visualization functions like :meth:`mne.io.Raw.plot`. They will be shown
        correctly in :func:`mne.viz.eyetracking.plot_gaze`. See :gh:`11879`
        for more information.
```

Reply:
> We should have a prominent warning at the top like:

Does the concern with `raw.plot()` stem from the underlying issue that the unit conversion is nonlinear? Because other than that, `raw.plot` with radian units might actually be nicer because the scalings for eyegaze channels are radians in that function!

Reply:
> Does the concern with raw.plot() stem from the underlying issue that the unit conversion is nonlinear? Because other than that, raw.plot with radian units might actually be nicer because the scalings for eyegaze channels are radians in that function!

No not quite -- have you tried `raw.plot` after calling your conversion function? I strongly suspect first that it won't label the units properly in the scale bar, probably still as `AU` (rather than pixels before or radians after) because of this for example:

https://github.com/mne-tools/mne-python/blob/566c6ea5a6c5c66657e9edcffb1e596d21ba6952/mne/defaults.py#L98

And second -- and maybe worse since AU is at least ambiguous -- the `scalings` will make it hard to see the traces because the default `scalings` for eye gaze is presumably set up to be reasonable for pixels (?) which means it accounts for a hundreds-to-about-a-thousand range. The 30 degree-ish max you talk about here is only ~0.52 radians, so I'm guessing the traces after conversion will look almost flat (to be confirmed!).
 
This has to do with the `FIFF_UNIT` stuff from https://github.com/mne-tools/mne-python/issues/11879, where all (other) plotting functions assume there is a static channel-type-to-unit mapping that is always the same. In principle we should use `info['chs'][ii]['unit']` or similar (and maybe to simplify things assume it's the same for all channels of a given type) but we don't. Hence why I think we should warn that other viz functions aren't built to handle conversion away from the native pixel units.

Reply:
> Hence why I think we should warn that other viz functions aren't built to handle conversion away from the native pixel units.

Ok Got it, I'll add the warning to the doc!



> have you tried `raw.plot` after calling your conversion function? I strongly suspect first that it won't label the units properly in the scale bar, probably still as `AU` ...  And second ... the `scalings` will make it hard to see the traces because the default `scalings` for eye gaze is presumably set up to be reasonable for pixels (?)

Yeah you are right that the units are always labeled as `AU`, but the default `scalings` for eye gaze are radians (as decided in [this comment](https://github.com/mne-tools/mne-python/pull/11152#issuecomment-1449102125) in PR #11152 ):

https://github.com/mne-tools/mne-python/blob/566c6ea5a6c5c66657e9edcffb1e596d21ba6952/mne/defaults.py#L101-L129



## Pixels

<details>
  <summary>Code</summary>
  
  ```
  import mne
  mne.viz.set_browser_backend("qt")
  
  fname = mne.datasets.eyelink.data_path() / "freeviewing" / "sub-01_task-freeview_eyetrack.asc"
  raw = mne.io.read_raw_eyelink(fname, create_annotations=["blinks"])

  raw.plot(theme="light", scalings=dict(eyegaze=1e3))
  ```
  
</details> 

<img src="https://github.com/mne-tools/mne-python/assets/52462026/c65cd73e-9b1e-4c29-aeed-16137b485293" width="600"/>
  

# Radians

<details>
  <summary>Code</summary>
  
  ```
  import mne
  mne.viz.set_browser_backend("qt")
  
  fname = mne.datasets.eyelink.data_path() / "freeviewing" / "sub-01_task-freeview_eyetrack.asc"
  raw = mne.io.read_raw_eyelink(fname, create_annotations=["blinks"])
  cal = mne.preprocessing.eyetracking.read_eyelink_calibration(fname,
                                                              screen_distance=.82,
                                                              screen_size=(.53, .3),
                                                              screen_resolution=(1920, 1080)
                                                              )[0]

 mne.preprocessing.eyetracking.convert_units(raw, cal, to="radians")
  

  raw.plot(theme="light", scalings=dict(eyegaze=1e3))
  ```
  
</details> 
<img src="https://github.com/mne-tools/mne-python/assets/52462026/5a0d9672-0716-4624-884a-4b74efb82734" width="600"/>


note that when eyegaze units are px, I explicitly set the scalings i.e. ` raw.plot(scalings=dict(eyegaze=1e3))`, whereas I don't need to explicitly set the scalings after converting the data to radians, and the traces look ok?

Reply:
Yeah I could have it backward -- just make sure the warning says that other plotting funcs won't work right for the unit that they won't (guess that's pixels)